Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -19,10 +20,22 @@ message EventCount{
     uint32 Value = 1;
 }
 
+message EventCountByTypeKey{
+    ExecutionEventId EventTypeID = 1;
+}
+
+message EventCountByTypeValue{
+    map<uint32, int32> WorkerToCountMap = 1;
+
+    map<uint32, int32> WorkerToPayloadMap = 2;
+}

```

Comment:
Explicitly storing the event counts in the db feels very specific for that one analyzer. Is it worth storing it in the db? It doesn't feel very generic.

Reply:
It is not generic, and it really just helps 1 analyzer, but at the same time, its 13 elements stored into a db so it is "cheap" and could fall under the "interesting indexing" for the db :) 

---
In reply to: [314450465](https://github.com/microsoft/BuildXL/pull/724#discussion_r314450465) [](ancestors = 314450465)

Reply:
I suspect we will get more examples where we can benefit from storing information in the database, either for a single analysis or for future. I.e. critical path computation for each node. Having that computed once and then stored int he db can benefit many analyzers.

Perhaps if we label these somehow as derived tables (either by name prefix, or placing them in a shared location, different assembly etc, don't have a good idea of what the best longterm model is for that.

We can set the trend and guidelines for others to follow.


---
In reply to: [314472749](https://github.com/microsoft/BuildXL/pull/724#discussion_r314472749) [](ancestors = 314472749,314450465)